services/cd-service/pkg/repository/repository.go

```diff
@@ -2362,6 +2363,12 @@ func (s *State) WriteAllReleases(ctx context.Context, transaction *sql.Tx, app s
 		if err != nil {
 			return fmt.Errorf("cannot get manifest for app %s and release %v: %v", app, releaseVersion, err)
 		}
+
+		if !valid.SHA1CommitID(repoRelease.SourceCommitId) {
+			logger.FromContext(ctx).Sugar().Warnf("Source commit ID %s is not valid. Skipping migration for release %d of app %s", repoRelease.SourceCommitId, releaseVersion, app)
+			continue
```

Comment:
I don't think we should skip the release, but just set the sourceCommitId to null.